### PR TITLE
Implement Pink Elephant

### LINKED ISSUE #556: Implement Pink Elephant
As follow-up for #521, since we have an accepted proposal for Pink Elephant, here is the implementation items for Pink Elephant (and more broadly any theme in the future)

 - [x] Create `src/theme_pink_elephant.scss`
    - This should be a copy of `src/theme_dark.scss` with line 13 changed to reflect Pink Elephant rather than dark theme
 - [x] Create `src/themes/oui/oui_colors_pink_elephant.scss`
    - This should be a copy of `src/themes/oui/oui_colors_dark.scss` for now. These colors will change in the future
 - [x] Create `src-docs/src/theme_pink_elephant.scss`
    - This should be a copy of `src-docs/src/theme_dark.scss` with line 16 changed to reflect Pink Elephant rather than dark theme
 - [x] Add Pink Elephant to the list of themes (`OUI_THEMES`) in `src/themes/themes.ts`
    - `text`: `Pink Elephant`
    - `value`: `pink-elephant`
 - [x] Register the theme in `src-docs/src/index.js`
    - Import the theme like the other themes: https://github.com/opensearch-project/oui/blob/f5d7aa66fa05e211f60b20cd26612472da9e5ee0/src-docs/src/index.js#L25-L28
    - Register the theme like the other themes: https://github.com/opensearch-project/oui/blob/f5d7aa66fa05e211f60b20cd26612472da9e5ee0/src-docs/src/index.js#L33-L36

The implementation of this task can be based off the work done here: https://github.com/opensearch-project/oui/compare/main...BSFishy:oui:pink_elephant

### PR DESCRIPTION
**Pull request by AbhishekReddy1127**
### Description
Implements the Pink Elephant and more broadly any theme in the future.

Screenshot (24)
 
### Issues Resolved
Fixes #556 
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [x] All tests pass
  - [x] `yarn lint`
  - [x] `yarn test-unit`
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/oui/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by seanneumann** [comment]
This is exciting. Can I see some screenshots when they're ready?

File: src/themes/oui/oui_colors_pink_elephant.scss [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ $euiColorChartLines: $ouiColorChartLines;
+ $euiColorChartBand: $ouiColorChartBand;
+ $euiShadowColor: $ouiShadowColor;
+ $euiShadowColorLarge: $ouiShadowColorLarge;
+ $euiBreadcrumbBlueBackground: $ouiBreadcrumbBlueBackground;
+ $euiBreadcrumbGrayBackground: $ouiBreadcrumbGrayBackground;
+ $euiBreadcrumbCollapsedLink: $ouiBreadcrumbCollapsedLink;
+ $euiBreadCrumbHoverColor: $ouiBreadCrumbHoverColor;
+ /* End of Aliases */
+
```
Comment (BSFishy) [inline src/themes/oui/oui_colors_pink_elephant.scss:111]: This whole file is indented with 1 space, which is causing build failure

**Review by BSFishy — CHANGES_REQUESTED** [review]
Please update theme colors to match https://github.com/opensearch-project/oui/issues/521#issuecomment-1451101640

File: src/themes/oui/oui_colors_pink_elephant.scss [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+@import '../../global_styling/functions/index';
+
+// These colors stay the same no matter the theme
+$ouiColorGhost: #FFF;
+$ouiColorInk: #4012BC;
```
Comment (BSFishy) [inline src/themes/oui/oui_colors_pink_elephant.scss:16]: This must remain at `#000`. Cannot change with theme

Comment (BSFishy) [reply src/themes/oui/oui_colors_pink_elephant.scss:16]: Maybe update `$ouiTextColor` instead?

File: src/themes/oui/oui_colors_pink_elephant.scss [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+$ouiColorEmptyShade: #FDDFEC;
+$ouiColorLightestShade: #FBD2E2;
+$ouiColorLightShade: #F6BFCB;
+$ouiColorMediumShade: #E0C3D3;
+$ouiColorDarkShade: #928EA2;
+$ouiColorDarkestShade: #5F5B6B;
+$ouiColorFullShade: #FFF;
+ 
+// Backgrounds
+$ouiPageBackgroundColor: shade($ouiColorLightestShade, 45%);
```
Comment (BSFishy) [inline src/themes/oui/oui_colors_pink_elephant.scss:38]: This should be
```scss
$ouiPageBackgroundColor: $ouiColorLightestShade;
```

The shade causes some absurd color shifting issues

**Review by BSFishy — CHANGES_REQUESTED** [review]

**Review by BSFishy — APPROVED** [review]

**Pull request merged**